packages/password/tests/apple.password.test.js

```diff
@@ -0,0 +1,61 @@
+const { internal } = require('../lib/apple.password')
+const fc = require('fast-check')
```

Comment:
Can you explain the advantage of using `fast-check`?

Reply:
Yeah, my API is designed to be extremely robust (internally), so `fast-check` is used in a few places to stress-test methods with various types of **auto-generated** test data.

For example, imagine you found a bug where passing `"foo"` to a method was fine, but passing  `""` (empty string) causes it to throw, or do something unexpected. You could start adding test cases like this in Jest:

```js
it.each([
    "food", // first test case  
    ""      // second case, empty string
])(input => {
    expect(myApi(input)).toBeDefined()
})
```

Where the value of this test is that you're sure for those 2 inputs, run-time exceptions will not occur.

The problem here is that we've only added 2 test cases, and we're not really asserting much, just that the method doesn't throw - which is absolutely a valuable test, but coming up with other known failures is hard/impossible.

What if another bug is found, say if you provide `"    "` (lots of spaces) - you could keep extending the test cases above, but you get to a point where you realise that what you really want is a way to **generate** test inputs, so that you can stress test methods without having to come up with the test cases yourself.

So with `fast-check`, (other property-based testing libraries are available), you can stress-test a method like so

```
it('should produce positive integers from ranges _randomNumberWithUniformDistribution', () => {
    fc.assert(
        fc.property(fc.integer({min: 1}), data => {
            const result = internal._randomNumberWithUniformDistribution(data)
            return result >= 0
        })
    )
})
``

The interesting part here is that `internal._randomNumberWithUniformDistribution` will be executed possibly hundreds of times there, and all this test cares about is that `result` is a positive integer. 

Not all methods are suited to this style, but since `fast-check` works seamlessly inside our existing Jest setup, it's a worthwhile addition to ensure critical APIs behave as we expect